File: Server/Resources/Product/ProductFilter.cs
using System;

namespace Server
{
    public class ProductFilter : Filter
    {
        public int[] Ids { get; set; }
        public string[] Names { get; set; }
        public Nullable<double> PriceMin { get; set; }
        public Nullable<double> PriceMax { get; set; }
        public BrandFilter BrandFilter { get; set; }
        public FormFactorFilter FormFactorFilter { get; set; }
    }
}

File: Client/src/app/shared/resources/product/product.filter.ts
import { BrandFilter } from "../brand/brand.filter";
import { Filter } from "../filter";
import { FormFactorFilter } from "../form-factor/form-factor.filter";

export class ProductFilter extends Filter {
    ids?: number[];
    names?: string[];
    priceMin?: number;
    priceMax?: number;
    brandFilter?: BrandFilter;
    formFactorFilter?: FormFactorFilter;

    constructor() {
        super();
        this.brandFilter = new BrandFilter();
        this.formFactorFilter = new FormFactorFilter();
    }
}